src/object/srv_ec.c

```diff
@@ -127,8 +127,19 @@ obj_ec_rw_req_split(daos_unit_oid_t oid, struct obj_iod_array *iod_array,
 
 	if (tgt_map != NULL) {
 		D_ASSERT(count == map_size);
+
+		/* If leader is not any EC shard, neither parity nor data,
+		 * then temporarily set leader as tgt_max_idx, that is not
+		 * important since current server will not take part in EC
+		 * update, the temporary leader is only for split handling.
+		 */
+		if (!obj_ec_is_valid_tgt(tgt_map, map_size, leader_id, &leader))
+			leader = tgt_max_idx;
```

Comment:
need it to add -
"else
        leader = leader_id;"?

Reply:
obj_ec_is_valid_tgt() will assign value to "leader" if it is in the tgt_map.